crates/examples/push-cdn/broker.rs

```diff
@@ -0,0 +1,93 @@
+//! The following is the main `Broker` binary, which just instantiates and runs
+//! a `Broker` object.
+
+use anyhow::{Context, Result};
+use cdn_broker::reexports::connection::protocols::{Quic, Tcp};
+use cdn_broker::{reexports::crypto::signature::KeyPair, Broker, Config, ConfigBuilder};
+use clap::Parser;
+use hotshot::traits::implementations::WrappedSignatureKey;
+use hotshot::types::SignatureKey;
+use hotshot_example_types::node_types::TestTypes;
+use hotshot_types::traits::node_implementation::NodeType;
+use local_ip_address::local_ip;
+
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+/// The main component of the push CDN.
+struct Args {
+    /// The discovery client endpoint (including scheme) to connect to
+    #[arg(short, long)]
+    discovery_endpoint: String,
+
+    /// Whether or not metric collection and serving is enabled
+    #[arg(long, default_value_t = true)]
+    metrics_enabled: bool,
+
+    /// The port to bind to for externalizing metrics
+    #[arg(long, default_value = "127.0.0.1")]
+    metrics_ip: String,
```

Comment:
I don't know if it matters, but these ip addresses aren't being validated, at least not where I can see. Not sure if that's something we care about.

Reply:
The broker will format these IPs on startup, and the error would be printed when we call `.start?`. Also just noticed the comment there is wrong